### PR TITLE
Poll until resource rename complete

### PR DESCRIPTION
**Pull request by markjamieson**
## Reason for change

Polls resource endpoint until renamed resource is returned, then emits success event.

## Testing

New to Render work, so there's possibly a better way, but I...
1. Built UI locally
1. Copied `dist/` to `<render-dashboard>/node_modules/@manifoldco/ui/`
1. Provisioned a resource
1. Renamed the resource

### CODE REVIEW

File: src/components/manifold-data-rename-button/manifold-data-rename-button.tsx [hunk 1]
```diff
@@ -126,6 +126,21 @@ export class ManifoldDataRenameButton {
       return Promise.reject(error);
     });
 
+    // Poll until rename complete
+    const restFetch = this.restFetch;
```
Comment (markjamieson) [inline src/components/manifold-data-rename-button/manifold-data-rename-button.tsx]: Does anybody know why I needed to do this? TS compiler did not like me trying to use `this.restFetch` inside this new Promise

File: src/components/manifold-data-rename-button/manifold-data-rename-button.tsx [hunk 2]
```diff
@@ -126,6 +126,21 @@ export class ManifoldDataRenameButton {
       return Promise.reject(error);
     });
 
+    // Poll until rename complete
+    const { restFetch } = this;
+    await new Promise(resolve => {
+      const interval = window.setInterval(async () => {
+        const renamedResource: [Marketplace.Resource] | undefined = await restFetch({
```
Comment (dangodev) [inline src/components/manifold-data-rename-button/manifold-data-rename-button.tsx]: Do you need a type: `this.restFetch<Marketplace.Resource>` ?

File: src/components/manifold-data-rename-button/manifold-data-rename-button.tsx [hunk 3]
```diff
@@ -126,6 +126,21 @@ export class ManifoldDataRenameButton {
       return Promise.reject(error);
     });
 
+    // Poll until rename complete
+    const { restFetch } = this;
```
Comment (markjamieson) [inline src/components/manifold-data-rename-button/manifold-data-rename-button.tsx]: I'm not really sure why this was necessary but TS didn't like me using `this.restFetch` inside this new Promise

File: src/components/manifold-data-rename-button/manifold-data-rename-button.tsx (hunk 2 above)
Comment (markjamieson) [reply src/components/manifold-data-rename-button/manifold-data-rename-button.tsx]: Yes, just fixed that up a little bit

File: src/components/manifold-data-rename-button/manifold-data-rename-button.tsx (hunk 3 above)
Comment (dangodev) [reply src/components/manifold-data-rename-button/manifold-data-rename-button.tsx]: Wonder if it’s a context thing 🤔

Comment (dangodev) [reply src/components/manifold-data-rename-button/manifold-data-rename-button.tsx]: Though seems like your solution still points to the same reference; it just hides the warning

**Pull request merged**